src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs

```diff
@@ -193,6 +194,7 @@ private static IList<Exception> Flatten(ICollection<Exception> exceptions)
             var aggregate = new AggregateException(exceptions);
             return aggregate.Flatten().InnerExceptions
                 .Select(UnwrapException)
+                .Where(ex => !(ex is JoinableTaskContextException))
```

Comment:
New exceptions are now thrown if you attempt to access services on the UI thread

```
Microsoft.VisualStudio.Threading.JoinableTaskContextException: An attempt to switch to the main thread failed to reach the expected thread. Was the JoinableTaskContext initialized on the wrong thread or with a SynchronizationContext whose Post method does not execute its delegate on the main thread?
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.MainThreadAwaiter.GetResult()
   at Microsoft.VisualStudio.Text.Outlining.OutliningManager.<SourceTextChangedAsync>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.<>c__DisplayClass41_0`1.<<RaiseEventOnBackgroundAsync>b__0>d.MoveNext()
```